ledger_hid.go

```diff
@@ -165,17 +166,52 @@ func (ledger *LedgerDeviceHID) readThread() {
 		buffer := make([]byte, PacketSize)
 		readBytes, err := ledger.device.Read(buffer)
 
+		// Check for HID Read Error (May occur even during normal runtime)
 		if err != nil {
-			return
+			continue
+		}
+
+		// Discard all zero packets from Ledger Nano X on macOS
+		allZeros := true
+		for i := 0; i < 64; i++ {
```

Comment:
maybe here would be better to write: 
`for i := 0; i < len(buffer);`

Reply:
updated